lib/collection/src/collection_manager/holders/segment_holder.rs

```diff
@@ -404,61 +405,95 @@ impl<'s> SegmentHolder {
+    /// selected to be updated, all older versions are marked to be deleted.
+    ///
+    /// Points that are already soft deleted are not included.
+    fn find_points_to_update_and_delete(
+        &self,
+        ids: &[PointIdType],
+    ) -> (
+        AHashMap<SegmentId, Vec<PointIdType>>,
+        AHashMap<SegmentId, Vec<PointIdType>>,
+    ) {
+        let mut to_delete: AHashMap<SegmentId, Vec<PointIdType>> = AHashMap::new();
+
+        // Find in which segments latest point versions are located, mark older points for deletion
+        let mut latest_points: AHashMap<PointIdType, (SeqNumberType, SegmentId)> =
             AHashMap::with_capacity(ids.len());
         for (segment_id, segment) in self.iter() {
             let segment_arc = segment.get();
             let segment_lock = segment_arc.read();
             let segment_points = Self::segment_points(ids, segment_lock.deref());
             for segment_point in segment_points {
-                if let Some(point_version) = segment_lock.point_version(segment_point) {
-                    match points.entry(segment_point) {
-                        // First time we see the point, add it to the list
-                        Entry::Vacant(entry) => {
-                            entry.insert((point_version, *segment_id));
-                        }
-                        // Point we have seen before is older, replace it
-                        Entry::Occupied(mut entry) if entry.get().0 < point_version => {
+                let Some(point_version) = segment_lock.point_version(segment_point) else {
+                    continue;
+                };
+
+                match latest_points.entry(segment_point) {
+                    // First time we see the point, add it
+                    Entry::Vacant(entry) => {
+                        entry.insert((point_version, *segment_id));
+                    }
+                    // Point we have seen before is older, replace it and mark older for deletion
+                    Entry::Occupied(mut entry) if entry.get().0 < point_version => {
+                        let (old_version, old_segment_id) =
                             entry.insert((point_version, *segment_id));
+
+                        // Mark other point for deletion if the version is older
+                        // TODO(timvisee): remove this check once deleting old points uses correct version
+                        if old_version < point_version {
+                            to_delete
+                                .entry(old_segment_id)
+                                .or_default()
+                                .push(segment_point);
                         }
-                        // Point we have seen before is newer, do nothing
-                        Entry::Occupied(_) => {}
+                    }
+                    // Ignore points with the same version, only update one of them
```

Comment:
That means that after update operation, one of the points became outdated, right?

Reply:
Correct. We changed this to update all points having the highest version.